controllers/storageautoscaler/scraper.go

```diff
@@ -107,16 +113,23 @@ func filterObjectsForScaling(storageAutoScalerList *ocsv1.StorageAutoScalerList,
 			err := fmt.Errorf("osd usage is nil for device class, device class: %s", storageAutoScaler.Spec.DeviceClass)
 			log.Error(err, "device class not found in sync map")
 		} else {
-			deviceClassUsage := usage.(float64)
-			if (deviceClassUsage * 100) >= float64(deviceClassThreshold) {
-				objectList = append(objectList, types.NamespacedName{Namespace: storageAutoScaler.Namespace, Name: storageAutoScaler.Name})
+			deviceClassUsage := usage.(OsdUsage)
+			if (deviceClassUsage.HighestUsage * 100) >= float64(deviceClassThreshold) {
+				if totalUsageGreaterThanThresholdMinusTenPercent(deviceClassUsage, deviceClassThreshold) {
+					objectList = append(objectList, types.NamespacedName{Namespace: storageAutoScaler.Namespace, Name: storageAutoScaler.Name})
+				}
+				log.Info("highest osd usage is greater than the threshold, but total cluster usage is less than the (threshold-10)%, skipping scaling", "device class", deviceClass, "highest osd usage", deviceClassUsage.HighestUsage*100, "total osd usage", deviceClassUsage.TotalUsage*100)
 			}
 		}
 	}
 
 	return objectList
 }
 
+func totalUsageGreaterThanThresholdMinusTenPercent(deviceClassUsage OsdUsage, deviceClassThreshold int) bool {
+	return (deviceClassUsage.TotalUsage/deviceClassUsage.TotalOsd)*100 >= math.Max(float64(deviceClassThreshold)-1, 0)
```

Comment:
Check if `TotalOsd` is 0, otherwise you could have divide by 0.

Reply:
Wait, we need a different computation. My comment [here](https://github.com/red-hat-storage/ocs-operator/pull/3200#discussion_r2080388652) must not be clear. The 10% difference to check is that if we expand horizontally, that we are expanding at least 10% of the capacity. If adding 3 OSDs will result in less than 10% expansion, then we need to have some multiple of 3 OSDs until it is at least 10%.

An exception to this rule would be if adding more than 3 OSDs will exceed the total autoscale storage limit, then we can just expand 3.

Reply:
we will do that process in the next release, currently sticking to add only 3 osds at a time

Reply:
Oh right, that will be a future PR. The calculation for this purpose looks good now.